packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx

```diff
@@ -20,6 +20,8 @@ import RedPacketDialog from './RedPacketDialog'
 import { RedPacketInPost } from './RedPacketInPost'
 import { RedPacketNftInPost } from './RedPacketNftInPost'
 import { RedPacketIcon, NFTRedPacketIcon } from '@masknet/icons'
+import { ApplicationEntry } from '@masknet/shared'
+import { requestComposition } from '@masknet/shared-base'
```

Comment:
`requreComposition` should not be an exported function in `'@masknet/shared-base'`.

It should be provided by the first argument of `init()`, and the plugin should not provide the plugin ID on it's own.

Reply:
Please add it in plugin context for every plugin.

Reply:
done